src/mongo/integration/standalone/dbclient_test.cpp

```diff
@@ -394,6 +394,67 @@ namespace {
         ASSERT_TRUE(saw_capped);
     }
 
+    TEST_F(DBClientTest, EnumerateCollections) {
+        const int n_collections = 9;
+        const int batch_size = 3;
+
+        for (int i = 0; i < n_collections; ++i) {
+            std::stringstream ss;
+            ss << TEST_DB + ".COLL" << i;
+            c.createCollection(ss.str());
+        }
+
+        BSONObjBuilder bob;
+        bob.appendRegex("name", "COLL\\d$");
+        auto_ptr<DBClientCursor> cursor = c.enumerateCollections(TEST_DB, bob.obj(), batch_size);
+
+        if (serverGTE(&c, 2, 8))
+            ASSERT_EQUALS(cursor->getns(), TEST_DB + ".$cmd.listCollections");
+        else
+            ASSERT_EQUALS(cursor->getns(), TEST_DB + ".system.namespaces");
+
+        int coll_count = 0;
+        while (cursor->more()) {
+            // The shims don't support the objsLeftInBatch cursor interface for initial batch.
```

Comment:
The way it was designed is that more() and next() are the only virtual parts of the DBClientCursor interface and the shims don't provide any cursor like behavior besides what is provided by those two methods. This has been the case since aggregation was first implemented, why do we want to change it now?